src/src/PluginStructs/P130_data_struct.cpp

```diff
@@ -0,0 +1,283 @@
+
+#ifdef USES_P130
+
+#define P130_ADS1X15_REG_POINTER_CONVERT         (0x00)
+#define P130_ADS1X15_REG_POINTER_CONFIG          (0x01)
+#define P130_ADS1X15_REG_POINTER_THRESH_LO       (0x02)
+#define P130_ADS1X15_REG_POINTER_THRESH_HI       (0x03)
+
+#define P130_ADS1015_MUX_DIFF_01            (0x0000)
+#define P130_ADS1015_MUX_DIFF_23            (0x3000)
+#define P130_ADS1015_PGA_1_024V             (0x0600) //< +/-1.024V range = Gain 4 => datasheet says 1bit = 0.5mV
+#define P130_ADS1015_CONFIG_CQUE_NONE       (0x0003) // Disable the comparator (default val)
+#define P130_ADS1015_CONFIG_CLAT_NONLAT     (0x0000) // Non-latching (default val)
+#define P130_ADS1015_CONFIG_CPOL_ACTVLOW    (0x0000) // Alert/Rdy active low   (default val)
+#define P130_ADS1015_CONFIG_CMODE_TRAD      (0x0000) // Traditional comparator (default val)
+#define P130_ADS1015_CONV_MODE_CONTINUOUS   (0x0000) // Single-shot mode (default)
+#define P130_ADS1015_CONV_MODE_SINGLE       (0x0100) // Single-shot mode (default)
+#define P130_ADS1015_START_SINGLE_CONV      (0x8000) // Start a single conversion
+
+P130_data_struct::P130_data_struct(uint8_t i2c_addr, uint8_t _calCurrent1, uint8_t _calCurrent2, float_t _calVoltage, uint8_t _currentFreq, uint8_t _nbSinus, uint8_t _convModeContinuous, uint8_t _sps) : 
+    i2cAddress(i2c_addr),
+    calCurrent1(_calCurrent1),
+    calCurrent2(_calCurrent2),
+    calVoltage(_calVoltage),
+    currentFreq(_currentFreq),
+    nbSinus(_nbSinus),
+    convModeContinuous (_convModeContinuous),
+    sps (_sps) {
+        this->debug = 0;
+    }
+
+void P130_data_struct::setDebug(uint8_t _debug) {
+    this->debug = _debug;
+    if (this->debug) {
+        String log;
+        log = F("Irms - ADS1015 : Set internal plugin debug; _calCurrent1=");
+        log += this->calCurrent1;
+        log += F(" _calCurrent2=");
+        log += this->calCurrent2;
+        log += F(" voltEst.=");
+        log += String(this->calVoltage);
+        log += F(" frequency=");
+        log += this->currentFreq;
+        log += F(" nbSinus=");
+        log += this->nbSinus;
+        log += F(" SPS=");
+        log += this->sps;
+        addLog(LOG_LEVEL_INFO, log);
+    }
+}
+
+uint8_t P130_data_struct::getDebug() {
+    return this->debug;
+}
+
+boolean P130_data_struct::writeRegister(uint8_t registerAddr, uint16_t registerValue) {
+    boolean result = false;
+    uint8_t res = 0;
+    Wire.beginTransmission(this->i2cAddress);
+    Wire.write((uint8_t)(registerAddr));
+    Wire.write((uint8_t)(registerValue >> 8));
+    Wire.write((uint8_t)(registerValue & 0xFF));
+    res = Wire.endTransmission();
+    if ( 0 == res ) {
+        result = true;
+    } else {
+        String log = F("Irms - ADS1015 : writeRegister NOK, errcode= ");
+        log += res ;
+        addLog(LOG_LEVEL_INFO, log);
+        result = false;
+    }
+    return result;
+}
+
+boolean P130_data_struct::readRegister(uint8_t registerAddr, uint16_t& registerValue) {

```

Comment:
Please use `I2C_read16_reg` in `ESPEasy/src/src/Helpers/I2C_access.h` instead of re-implementing those I2C functions. (same for `writeRegister` above)

Thing is, there is lots and lots of I2C code doing it wrong (e.g. calling `Wire.endTransmission()` after `Wire.requestForm()`) and when you're using a single function like the one I linked, I know it is correct and if it isn't, I only need to fix it at one place.